chore(flags): Add scheduled Celery task for HyperCache verification

haacked opened the pull request:
Adds an hourly task that verifies all teams' HyperCache entries (flags and team metadata) against the database and automatically fixes any issues found.

## Problem

We're attempting to keep a 100% warmed HyperCache for team metadata and flags. But every time I run the verification command, we find out of sync entries in the team metadata cache. Its unclear why, but until we track that down, we need a process that self-heals the cache.

## Changes

- Add `verify_and_fix_hypercaches_task` that runs hourly at minute 30, verifying both `team_metadata` and `flags` caches in sequence
- Extract verification logic to reusable functions:
  - `verify_team_flags()` and `verify_team_metadata()` can be called by both management commands and the scheduled task
  - Both functions now use `batch_data` when available to avoid N+1 queries
- Extract `batch_check_expiry_tracking()` to `hypercache_manager.py` for reuse by both the management commands and verification task
- Simplify management commands to delegate to the new verify functions
- Add `VerificationResult.formatted_fixed_team_ids()` to cap logged team IDs at first 10 with "... and N more" summary

Metrics: `posthog_hypercache_verify_fixes_total{cache_type, issue_type}`

## How did you test this code?

Unit Tests
Manual Testing

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Changelog: (features only) Is this feature complete?

<!-- Yes if this is okay to go in the changelog. No if it's still hidden behind a feature flag, or part of a feature that's not complete yet, etc.  -->
<!-- Removing this section does not mean the changelog bot won't pick it up, because *some people* like to not use the template, so we can't rely on it existing. -->

haacked committed:
Add scheduled Celery task for HyperCache verification

Adds an hourly task that verifies all teams' HyperCache entries (flags and
team metadata) against the database and automatically fixes any issues found.

Key changes:

- Add `verify_and_fix_hypercaches_task` that runs hourly at minute 30,
  verifying both team_metadata and flags caches in sequence

- Extract verification logic to reusable functions:
  - `verify_team_flags()` and `verify_team_metadata()` can be called by
    both management commands and the scheduled task
  - Both functions now use batch_data when available to avoid N+1 queries

- Extract `batch_check_expiry_tracking()` to `hypercache_manager.py` for
  reuse by both the management commands and verification task

- Simplify management commands to delegate to the new verify functions

- Add `VerificationResult.formatted_fixed_team_ids()` to cap logged team
  IDs at first 10 with "... and N more" summary

Metrics: posthog_hypercache_verify_fixes_total{cache_type, issue_type}

haacked committed:
Add tests for HyperCache verification utilities and Celery task

- Add tests for VerificationResult dataclass and formatted_fixed_team_ids()
- Add tests for _fix_and_record helper function
- Add tests for _verify_and_fix_batch batch processing
- Add tests for verify_and_fix_all_teams aggregation
- Add tests for verify_and_fix_hypercaches_task error handling

haacked committed:
Use None instead of empty string for issue field when status is match

This clearly indicates the absence of an issue rather than an empty string.

github-actions[bot] committed:
test(backend): update query snapshots

dmarticus reviewed (approved):
Approved with some minor feedback; up to your discretion. Loved the reusable verification logic.

haacked committed:
Address review feedback for HyperCache verification PR

- Add verbose parameter to verify_team_flags() and verify_team_metadata()
  to restore detailed diff output for management commands
- Reduce task time limit from 70 to 60 minutes to prevent overlapping
  hourly executions
- Make batch chunk size configurable via HYPERCACHE_VERIFICATION_CHUNK_SIZE
  environment variable (default: 1000)

haacked commented on posthog/models/feature_flag/flags_cache.py:
```diff
@@ -202,6 +202,86 @@ def update_flags_cache(team: Team | int, ttl: int | None = None) -> bool:
+                diffs.append({"type": "FIELD_MISMATCH", "flag_id": flag_id})
+
+    if not diffs:
+        return {"status": "match", "issue": None, "details": ""}
```
The issue is not read when "status" is "match", but I agree that I shouldn't have changed this as it's not relevant to the goals of this PR.
